### PR TITLE
Validation specs

### PR DESCRIPTION
**Pull request by kMutagene**
This PR will formalize the mechanisms discussed to bundle validation results in the ARC. Currently WIP, but happy for comments

cc @HLWeil @muehlhaus @omaus @j-bauer @Brilator

### CODE REVIEW

File: ARC specification.md [hunk 1]
```diff
@@ -298,9 +302,19 @@ Reproducibility of ARCs refers mainly to its *runs*. Within an ARC, it MUST be p
 
 ## Mechanism for Quality Control of ARCs
 
-ARCs are supposed to be living research objects and are as such never complete. Nevertheless, a mechanism to report the current state and quality of an ARC is indispensable. Therefore, ARCs will be scored according to the amount of metadata information available (specifically with established minimal metadata standards such as MinSeqE, MIAPPE, etc.), the quality of data and metadata (this metric will be established in the next version), manual curation and review, and the reuse of ARCs by other researchers measured by physical includes of the data and referencing.
+ARCs are supposed to be living research objects and are as such never complete. Nevertheless, a mechanism to continuously report the current state and quality of an ARC is indispensable. This process is further referred to as _validation_ of the ARC against a _target_, where the _target_ is a arbitrary set of validation cases that the ARC MUST pass to qualify as _valid_ in regard to the _target_. A reference implementation of a framework to create and run targets for ARC validation is provided in the [arc-validate repository]().
```
Comment (Brilator) [inline ARC specification.md:305]: an arbitrary

File: ARC specification.md [hunk 2]
```diff
@@ -298,9 +302,19 @@ Reproducibility of ARCs refers mainly to its *runs*. Within an ARC, it MUST be p
...
 
-To foster FAIRification, badges will be earned by reaching certain scores for a transparent review process.  
+ARCs MAY be validated against 0 or more targets defined in a [validation_targets.yml file](#structure-of-the-validation_targetsyml-file), where the folowing criteria MUST be met for each target:
```
Comment (Brilator) [inline ARC specification.md:307]: following

**Comment by Brilator** [comment]
Looks good. Just wondering what this looks like for ARCs being validated against multiple targets (e.g. invenio and pride) especially in terms of pducuding one badge.svg

**Comment by kMutagene** [comment]
> Looks good. Just wondering what this looks like for ARCs being validated against multiple targets (e.g. invenio and pride) especially in terms of pducuding one badge.svg

Ill go into detail of that when i work on the branch structure section

**Comment by kMutagene** [comment]
I think this is ready for discussion and review @HLWeil @Brilator

File: ARC specification.md [hunk 3]
```diff
@@ -298,9 +302,90 @@ Reproducibility of ARCs refers mainly to its *runs*. Within an ARC, it MUST be p
...
+  - `{$branch}` is the name of the branch the validation was run on
+  - `{$commithash}` is the full hash of the commit the validation was run on. See 
+  - `{$target}` is the name of the target the validation was run against. This folder then MUST contain the files `validation_report.*` and `badge.svg` as described [above](#mechanism-for-quality-control-of-arcs).
+
+  example:
+
+  > This example shows the validation results of the `main` and `branch-1` branches of the ARC repository against the `target1` and `target2` targets for two commits per branch:
+
+  ```
+  validation-branch-root
```
Comment (HLWeil) [inline ARC specification.md:341]: beautiful <3

File: ARC specification.md [hunk 4]
```diff
@@ -298,9 +302,90 @@ Reproducibility of ARCs refers mainly to its *runs*. Within an ARC, it MUST be p
...
+
+ARCs MAY be validated against 0 or more targets defined in a [validation_targets.yml file](#structure-of-the-validation_targetsyml-file), where the following criteria MUST be met for each target:
+- the target MUST have a unique _name_ across all validation targets used the ARC. This name MUST used for specifying the target in the [validation_targets.yml file](#structure-of-the-validation_targetsyml-file) and the subfolder names in the [validation branch](#structure-of-the-validation-branch)
+- target MUST create a `validation_report.*` file that summarizes the results of validating the ARC against the cases defined in the target. The format of this file SHOULD be of an established test result format such as [JUnit XML](https://github.com/windyroad/JUnit-Schema) or [TAP](https://testanything.org/).
```
[the remark this answers is not in the document]
Comment (kMutagene) [reply ARC specification.md]: clarified in next commit, now specifies that a validation package must be executable, e.g. as a script

File: ARC specification.md [hunk 5]
```diff
@@ -298,9 +302,90 @@ Reproducibility of ARCs refers mainly to its *runs*. Within an ARC, it MUST be p
...
-
-To foster FAIRification, badges will be earned by reaching certain scores for a transparent review process.  
+ARCs are supposed to be living research objects and are as such never complete. Nevertheless, a mechanism to continuously report the current state and quality of an ARC is indispensable. This process is further referred to as _validation_ of the ARC against a _target_, where the _target_ is an arbitrary set of validation cases that the ARC MUST pass to qualify as _valid_ in regard to the _target_. A reference implementation of a framework to create and run targets for ARC validation is provided in the [arc-validate repository](https://github.com/nfdi4plants/arc-validate).
+
+ARCs MAY be validated against 0 or more targets defined in a [validation_targets.yml file](#structure-of-the-validation_targetsyml-file), where the following criteria MUST be met for each target:
+- the target MUST have a unique _name_ across all validation targets used the ARC. This name MUST used for specifying the target in the [validation_targets.yml file](#structure-of-the-validation_targetsyml-file) and the subfolder names in the [validation branch](#structure-of-the-validation-branch)
```
[the remark this answers is not in the document]
Comment (kMutagene) [reply ARC specification.md]: I hope this is more clear with the next commit, which adds separation between validation and continuous quality control concepts

File: ARC specification.md [hunk 6]
```diff
@@ -298,9 +302,90 @@ Reproducibility of ARCs refers mainly to its *runs*. Within an ARC, it MUST be p
...
-
-To foster FAIRification, badges will be earned by reaching certain scores for a transparent review process.  
+ARCs are supposed to be living research objects and are as such never complete. Nevertheless, a mechanism to continuously report the current state and quality of an ARC is indispensable. This process is further referred to as _validation_ of the ARC against a _target_, where the _target_ is an arbitrary set of validation cases that the ARC MUST pass to qualify as _valid_ in regard to the _target_. A reference implementation of a framework to create and run targets for ARC validation is provided in the [arc-validate repository](https://github.com/nfdi4plants/arc-validate).
```
[the remark this answers is not in the document]
Comment (kMutagene) [reply ARC specification.md]: i think this is also adressed in the next iteration

File: ARC specification.md [hunk 7]
```diff
@@ -296,102 +302,156 @@ Notes:
...
-  - target2
+validation_packages:
+  - package1
+  - package2
 ```
 
-## Best Practices
+### Reference implementation
+
+PLANTDataHUB performs Continuous Quality Control of ARCs using the [arc-validate software suit]() as described in our 2023 paper [PLANTdataHUB: a collaborative platform for continuous FAIR data sharing in plant research](https://doi.org/10.1111/tpj.16474).
```
Comment (Brilator) [inline ARC specification.md:448]: link missing [arc-validate software suit](https://github.com/nfdi4plants/arc-validate)

**Comment by Brilator** [comment]
software suit => software suite

File: ARC specification.md [hunk 8]
```diff
@@ -296,102 +302,156 @@ Notes:
...
+
+  Any ARC where all sample names are prefixed with the string "Sample_" would be considered valid in regard to this validation case.
+
+### Validation packages
+
+A **validation package** bundles a collection of [validation cases](#validation-cases) that the ARC MUST pass to qualify as _valid_ in regard to the _validation package_ with instructions on how to perform the validation and summarize the results.
+
+Validation packages
+
+- MUST be executable.
```
Comment (HLWeil) [inline ARC specification.md:341]: So validation cases don't need to be executable (see "no further requirements are made on the format of validation cases") but the packages do?

File: ARC specification.md [hunk 9]
```diff
@@ -296,102 +302,156 @@ Notes:
...
+
+- MUST have a globally unique name.
+  This will eventually be enforced by a central validation package registry
+
+- SHOULD be versioned using [semantic versioning](https://semver.org/)
+
+- MUST create a `validation_report.*` file upon execution that summarizes the results of validating the ARC against the cases defined in the validation package.
+  The format of this file SHOULD be of an established test result format such as [JUnit XML](https://github.com/windyroad/JUnit-Schema) or [TAP](https://testanything.org/).
+
+- MUST create a `badge.svg` file upon execution that visually summarizes the results of validating the ARC against the validation cases defined in the validation package.
```
Comment (HLWeil) [inline ARC specification.md:354]: Any requirements of the shape of the contained visual summary, (e.g. according to badge.io formats)?

File: ARC specification.md [hunk 10]
```diff
@@ -296,102 +302,156 @@ Notes:
...
+A reference implementation for creating validation cases, validation packages, and validating ARCs against them is provided in the [arc-validate software suit](https://github.com/nfdi4plants/arc-validate)
+
+## Continuous quality control
+
+In addition to manually validate ARCs against validation packages, ARCs MAY be continuously validated against validation packages using a continuous integration (CI) system. 
+This process is further referred to as _Continuous Quality Control_ (CQC) of the ARC. CQC can be triggered by any event that is supported by the CI system, e.g. a push to a branch of the ARC repository or a pull request.
+
+### The cqc branch
+
+To make sure that validation results are bundled with the ARC but do not pollute their commit history, validation results MUST be stored in a separate branch of the ARC repository.
```
Comment (HLWeil) [inline ARC specification.md:368]: pollute `its`  commit history

File: ARC specification.md [hunk 11]
```diff
@@ -296,102 +302,156 @@ Notes:
...
               └── validation_report.xml
   ```
 
-### Structure of the validation_targets.yml file
+### The validation_packages.yml file
 
-The `validation_targets.yml` specifies the targets that the branch containing the file will be validated against. Each branch of an ARC MAY contain 0 or 1 `validation_targets.yml` file. If the file is present, it:
-  - MUST be located in an `.arc` folder in the root of the ARC
-  - MUST contain the `targets` key which is a list of target names that the current branch will be validated against.
+The `validation_packages.yml` specifies the validation packages that the branch containing the file will be validated against.
```
Comment (HLWeil) [inline ARC specification.md:430]: validated against `in the cqc`.
?

**Review by HLWeil — COMMENTED** [review]
Much better!

File: ARC specification.md (hunk 9 above)
Comment (kMutagene) [reply ARC specification.md:354]: You mean shields.io? no i think as long as this is a svg everything is fine

File: ARC specification.md (hunk 8 above)
Comment (kMutagene) [reply ARC specification.md:341]: yes

File: ARC specification.md (hunk 10 above)
Comment (kMutagene) [reply ARC specification.md:368]: i think ill go for plural in both cases, 
> To make sure that validation results are bundled with ARCs but do not pollute their commit history

File: ARC specification.md (hunk 11 above)
Comment (kMutagene) [reply ARC specification.md:430]: not necessarily. a local tool can take information from here as well

File: ARC specification.md (hunk 9 above)
Comment (HLWeil) [reply ARC specification.md:354]: Ah yes 😅 
Alright

**Review by HLWeil — APPROVED** [review]
lgtm

**Pull request merged**